staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr_test.go

```diff
@@ -110,6 +110,74 @@ func TestIntOrStringMarshalJSONUnmarshalYAML(t *testing.T) {
 	}
 }
 
+func TestGetIntFromString(t *testing.T) {
+	tests := []struct {
+		input      IntOrString
+		expectErr  bool
+		expectVal  int
+		expectPerc bool
+	}{
+		{
+			input:      FromInt(200),
+			expectErr:  false,
+			expectVal:  200,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("200"),
+			expectErr:  false,
+			expectVal:  200,
+			expectPerc: false,
+		},
+		{
+			input:      FromString("30%0"),
```

Comment:
would also be good to include "40%%" here

Reply:
added the test